### DISCUSSION TITLE
Mysql 에서 플러그인끼리 통신을 하는 이유는?

### DISCUSSION THREAD

**Discussion by Soyflat** [Well-Founded-Confidence/RealMySQL#29] [Q&A]
MySQL에서 플러그인끼리 통신을 하는 이유는 다양합니다. 여기에는 다음과 같은 몇 가지 이유가 있습니다.

첫째, MySQL 서버가 `대량의 데이터를 처리해야 하는 경우, 플러그인을 사용하여 복잡한 작업을 수행`할 수 있습니다. 예를 들어, 플러그인은 새로운 함수를 제공하거나 복잡한 계산을 수행할 수 있습니다.

둘째, MySQL 서버는 많은 클라이언트 요청을 처리해야 합니다. 플러그인은 새로운 프로토콜을 제공하여 클라이언트와 통신할 수 있습니다. 예를 들어, 플러그인은 새로운 프로토콜을 통해 클라이언트 요청을 처리하고, 처리된 결과를 MySQL 서버로 전달할 수 있습니다.

셋째, MySQL 서버는 보안 요구 사항을 충족해야 합니다. 플러그인은 새로운 인증 및 암호화 방법을 제공할 수 있습니다. 예를 들어, 플러그인은 새로운 인증 방식을 제공하여 사용자 인증을 처리하거나, 데이터베이스 암호화를 제공할 수 있습니다.

넷째, MySQL 서버는 다양한 운영 체제에서 실행됩니다. 플러그인은 운영 체제 간에 호환성을 제공하고, 운영 체제별로 서로 다른 작업을 수행할 수 있습니다.

따라서, 플러그인끼리 통신을 하는 것은 MySQL 서버의 기능을 확장하고, 복잡한 작업을 수행하고, 다양한 운영 체제 및 보안 요구 사항을 충족하는 데 필요한 유연성을 제공합니다.

**Comment by menthamin** [comment]
플러그인들 간에 통신을 해서 복잡한 일을 처리할 수 있다.
플러그인 1 -> 플러그인 2 -> 복잡한 결과 (좋은)
